### PR TITLE
kots remove command for apps

### PR DESCRIPTION
**Pull request by divolgin**
```
kubectl kots remove myapp-1 -n default
  • Removing application myapp-1
  • Application myapp-1 has been removed
```
```
kubectl kots remove myapp-1 -n default
  • Removing application myapp-1
Error: app with slug myapp-1 not found
```
```
kubectl kots remove myapp -n default
  • Removing application myapp
Error: failed to remove app, command may be retried with --force flag: version 0 is deployed
```
```
kubectl kots remove myapp -n default --force
  • Removing application myapp
  • Application myapp has been removed
```

### CODE REVIEW

File: kotsadm/pkg/store/s3pg/app_store.go [hunk 1]
```diff
@@ -369,3 +369,73 @@ func (c S3PGStore) SetSnapshotSchedule(appID string, snapshotSchedule string) er
...
+	db := persistence.MustGetPGSession()
+	tx, err := db.Begin()
+	if err != nil {
+		return errors.Wrap(err, "failed to begin transaction")
+	}
+	defer tx.Rollback()
+
+	var query string
+
+	// TODO: api_task_status needs app ID
```
Comment (salahalsaleh) [inline kotsadm/pkg/store/s3pg/app_store.go:386]: the function does have an app id though?

**Review by salahalsaleh — APPROVED** [review]

File: cmd/kots/cli/remove.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+			if err != nil {
+				log.FinishSpinnerWithError()
+				return errors.Wrap(err, "failed to read server response")
+			}
+
+			type removeAppResponse struct {
+				Error string `json:"error"`
+			}
+			response := removeAppResponse{}
+			_ = json.Unmarshal(b, &response)
```
[the remark this answers is not in the document]
Comment (divolgin) [reply cmd/kots/cli/remove.go:115]: what to do with it though?  I don't want it printed because there is nothing the user can do about it.

**Comment by dexhorthy** [comment]
whoah! Nice!

**Comment by markpundsack** [comment]
@divolgin This looks pretty good, but I'd like to see a couple things.

1. We should be clear what this is doing. It's good that it's not called `delete`, but still `remove` is ambiguous. Can we explicitly say it removes the reference or records of the app, but leaves the app objects intact?

2. In the error case, can you describe what went wrong? Being able to `--force` is nice, but as a user, I'd have no idea what force is going to do that regular `remove` wouldn't do. Usually `--force` is used to override some safety check.

3. What the heck does `version 0 is deployed` mean? Does that mean that version was already deployed, and that somehow is why we can't remove it? Or we failed to remove it so we redeployed version 0?

**Pull request merged**